js/Field.js

```diff
@@ -0,0 +1,212 @@
+
+            if ( !this.isAnyMineHere(y,x) ) {
+                this._table.rows[y].cells[x].dataset.info = 'M';
+                i++;
+            }
+        }
+        this._showMinesLeft();
+    }
+    _setNumbers() {
+        for(let i = 0; i<this.height; i++) {
+            for (let j = 0; j < this.width; j++) {
+                if ( !this.isAnyMineHere(i,j) ) {
+                    let count = this.mineCounter( this.cellNeighbours(i,j) );
+
+                    this._table.rows[i].cells[j].dataset.info = count || '';
+                }
+            }
+        }
+    }
+
+    _openCell(event) {
+        let td = event.target.closest('td');
+        if (!td) return null;
+        let y = td.parentNode.rowIndex;
+        let x = td.cellIndex;
+
+        if ( this.isAnyMineHere(y, x) ) {
+            this._endOfGame('gameOver');
+            return null;
+        }
+
+        td.dataset.type = 'opened';
+        this._showOpenedCell(td);
+
+        if ( td.dataset.info ==='' ) {
+            this._openNeighbours(y,x);
+        }
+
+        if ( this._doesPlayerWin() ) {
+            this._endOfGame('victory');
+        }
+    }
+    _setFlag(event) {
+        event.preventDefault();
+        let td = event.target.closest('td');
+        if (!td) return null;
+
+        let tdType = td.dataset.type;
+        if (tdType !== 'covered' && tdType !== 'marked') {
+            return null;
+        }
+
+        switch (tdType) {
+            case 'covered': {
+                td.dataset.type = 'marked';
+                this._flags++;
+                break;
+            }
+                case 'marked': {
+                    td.dataset.type = 'covered';
+                    this._flags--;
+                    break;
+                }
+        }
+
+        this._toggleFlag(td);
+        this._showMinesLeft();
+    }
+    _openNeighbours(y,x) {
+        this.cellNeighbours(y, x).forEach((td)=>{
+            let isMine = this.isAnyMineHere(td.parentNode.rowIndex, td.cellIndex);
+            let isOpened = (td.dataset.type ==='opened');
+            if ( !(isMine || isOpened) ) {
+                let event = new MouseEvent("click", {
+                    bubbles: true,
+                    cancelable: true
+                });
+                td.dispatchEvent(event);
+            }
+        })
+    }
+    _doesPlayerWin() {
+        let victory = true;
+        for(let i = 0; i<this.height; i++) {
+            for (let j = 0; j < this.width; j++) {
+                let td = this._table.rows[i].cells[j];
+                let isMine = this.isAnyMineHere(i, j);
+                let isClosed = (td.dataset.type ==='covered' || td.dataset.type ==='marked');
+                if (isClosed && !isMine) {
+                    victory = false;
+                    break;
+                }
+            }
+        }
+        return victory;
+    }
+    _endOfGame(typeOfEnd) {
+        this._table.removeEventListener('click', this._openCell);
```

Comment:
When I was adding 'simultaneous click' I undestood that you're right. So it is fixed